core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java

```diff
@@ -19,115 +19,299 @@
+    public static final String DEFAULT_OPTIMIZE_BBOX = "memory";
+    /** Default for coercing lon/lat values to a standard coordinate system */
+    public static final boolean DEFAULT_COERCE = false;
+    /** Default for accepting accept geo points with invalid latitude or longitude */
+    public static final boolean DEFAULT_IGNORE_MALFORMED = false;
 
-    private final String name;
-
-    private String distance;
-
-    private double lat;
-
-    private double lon;
+    private final String fieldName;
+    /** Distance from center to cover. */
+    private double distance;
+    /** Point to use as center. */
+    private GeoPoint center = new GeoPoint(Double.NaN, Double.NaN);
+    /** Algorithm to use for distance computation. */
+    private GeoDistance geoDistance = DEFAULT_GEO_DISTANCE;
+    /** Whether or not to use a bbox for pre-filtering. TODO change to enum? */
+    private String optimizeBbox = DEFAULT_OPTIMIZE_BBOX;
+    /** Whether or not to normalize longitude and latitude values to a standard coordinate system */
+    private boolean coerce = DEFAULT_COERCE;
+    /** Whether or not to accept geo points with invalid latitude or longitude */
+    private boolean ignoreMalformed = DEFAULT_IGNORE_MALFORMED;
 
-    private String geohash;
+    static final GeoDistanceQueryBuilder PROTOTYPE = new GeoDistanceQueryBuilder();
 
-    private GeoDistance geoDistance;
-
-    private String optimizeBbox;
-
-    static final GeoDistanceQueryBuilder PROTOTYPE = new GeoDistanceQueryBuilder(null);
-
-    private Boolean coerce;
+    /** For serialization purposes only. */
+    private GeoDistanceQueryBuilder() {
+        this.fieldName = null;
+    }
 
-    private Boolean ignoreMalformed;
+    /**
+     * Construct new GeoDistanceQueryBuilder.
+     * @param fieldName name of indexed geo field to operate distance computation on.
+     * */
+    public GeoDistanceQueryBuilder(String fieldName) {
+        Preconditions.checkArgument(Strings.isEmpty(fieldName) == false);
+        this.fieldName = fieldName;
+    }
 
-    public GeoDistanceQueryBuilder(String name) {
-        this.name = name;
+    /** Name of the field this query is operating on. */
+    public String fieldName() {
+        return this.fieldName;
     }
 
-    public GeoDistanceQueryBuilder point(double lat, double lon) {
-        this.lat = lat;
-        this.lon = lon;
+    /** Sets the center point for the query.
+     * @param point the center of the query
+     **/
+    public GeoDistanceQueryBuilder point(GeoPoint point) {
+        this.center = point;
         return this;
     }
 
-    public GeoDistanceQueryBuilder lat(double lat) {
-        this.lat = lat;
+    /**
+     * Sets the center point of the query.
+     * @param lat latitude of center
+     * @param lon longitude of center
+     * */
+    public GeoDistanceQueryBuilder point(double lat, double lon) {
+        this.center = new GeoPoint(lat, lon);
```

Comment:
Not sure either, if there is a reason to avoid object creation here I can sure change it. See that other parts in code use reset a lot, happy either way, wdyt?